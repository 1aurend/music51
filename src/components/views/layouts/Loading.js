import React, { useEffect, useState } from 'react'
import { Container, Row, Col } from 'shards-react'
import EndOfRound from '../../logic/EndOfRound'

export default function Loading({round}) {
  const [first, setFirst] = useState(true)
  useEffect(() => {
      window.scrollTo(0, 0)
      if (round !== 1) {
        setTimeout(() => setFirst(false), 100)
        return
      }
      setFirst(false)
  },[round])

  if (first) {
    return (
      <Container fluid className="main-content-container px-4" id='container'style={{backgroundColor: 'black', minHeight: '120vh'}}>
        <Row style={{paddingTop: '25%', justifyContent: 'center', marginLeft: '5%', marginRight: '5%'}}></Row>
          <Col sm='12' lg='12'>
            <Row style={{display: 'flex', justifyContent: 'center', marginLeft: '5%', marginRight: '5%'}} noGutters>
              <h2 style={{color: '#17c671', fontFamily: "'Press Start 2P', cursive"}}>Calculating your stats...</h2>
            </Row>
          </Col>
        </Container>
    )
  } else {
    return <EndOfRound round={round} />
  }

}
