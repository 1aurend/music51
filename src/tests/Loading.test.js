import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Loading from '../components/views/layouts/Loading'

test('loads with correct text', () => {
  const { getByRole, asFragment } = render(
  <Loading />
)
  expect(getByRole('heading')).toHaveTextContent('Calculating your stats...')
  expect(asFragment()).toMatchSnapshot()
})
