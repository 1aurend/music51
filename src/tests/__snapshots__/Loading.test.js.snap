// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads with correct text 1`] = `
<DocumentFragment>
  <div
    class="main-content-container px-4 container-fluid"
    id="container"
    style="background-color: black; min-height: 120vh;"
  >
    <div
      class="row"
      style="padding-top: 25%; justify-content: center; margin-left: 5%; margin-right: 5%;"
    />
    <div
      class="col-sm-12 col-lg-12"
    >
      <div
        class="no-gutters row"
        style="display: flex; justify-content: center; margin-left: 5%; margin-right: 5%;"
      >
        <h2
          style="color: rgb(23, 198, 113); font-family: 'Press Start 2P', cursive;"
        >
          Calculating your stats...
        </h2>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
